File: backend/SmartHomeApi/Api_Source_Code/Models/Data_Transaction/Devices/SmartDevices.cs
namespace SmartHomeApi.Api_Source_Code.Models;

public class SmartDevices{
    public List<Thermostat> Thermostats{get;set;}
    public List<SmartBulb> SmartBulbs{get;set;}
    public List<SmartJalousine> SmartJalousines{get;set;}
    public SmartDevices(List<Thermostat> thermostats, List<SmartBulb> smartBulbs, List<SmartJalousine> smartJalousines){
        this.Thermostats = thermostats;
        this.SmartBulbs = smartBulbs;
        this.SmartJalousines = smartJalousines;
    }
}

File: frontend/owasp-top10/src/app/models/database/SmartDevices.ts
import { SmartBulb } from "./Light/SmartBulb";
import { SmartJalousine } from "./Shutter/SmartJalousine";
import { Thermostat } from "./Temperature/Thermostat";

export interface SmartDevices {
    thermostats: Thermostat[],
    smartBulbs: SmartBulb[],
    smartJalousines: SmartJalousine[],
}